seam/utils/auth.py

```diff
@@ -0,0 +1,102 @@
+    is_publishable_key,
+    is_seam_token,
+    token_prefix,
+    access_token_prefix,
+)
+
+
+class SeamHttpInvalidTokenError(Exception):
+    def __init__(self, message):
+        super().__init__(f"SeamHttp received an invalid token: {message}")
+
+
+def get_auth_headers(
+    api_key: Optional[str] = None,
+    personal_access_token: Optional[str] = None,
+    workspace_id: Optional[str] = None,
+):
+    if is_seam_http_options_with_api_key(
+        api_key=api_key,
+        personal_access_token=personal_access_token,
+    ):
+        return get_auth_headers_for_api_key(api_key)
+
+    if is_seam_http_options_with_personal_access_token(
+        personal_access_token=personal_access_token,
+        api_key=api_key,
+        workspace_id=workspace_id,
+    ):
+        return get_auth_headers_for_personal_access_token(
+            personal_access_token, workspace_id
+        )
+
+    raise SeamHttpInvalidOptionsError(
+        "Must specify an api_key or personal_access_token. "
+        "Attempted reading configuration from the environment, "
+        "but the environment variable SEAM_API_KEY is not set."
+    )
+
+
+def get_auth_headers_for_api_key(api_key: str) -> dict:
+    if is_client_session_token(api_key):
+        raise SeamHttpInvalidTokenError(
+            "A Client Session Token cannot be used as an api_key"
+        )
+
+    if is_jwt(api_key):
+        raise SeamHttpInvalidTokenError("A JWT cannot be used as an api_key")
+
+    if is_access_token(api_key):
+        raise SeamHttpInvalidTokenError("An Access Token cannot be used as an api_key")
+
+    if is_publishable_key(api_key):
+        raise SeamHttpInvalidTokenError(
+            "A Publishable Key cannot be used as an api_key"
+        )
+
+    if not is_seam_token(api_key):
+        raise SeamHttpInvalidTokenError(
+            f"Unknown or invalid api_key format, expected token to start with {token_prefix}"
+        )
+
+    return {"authorization": f"Bearer {api_key}"}
+
+
+def get_auth_headers_for_personal_access_token(
+    personal_access_token: str, workspace_id: Optional[str] = None
+) -> dict:
+    if is_jwt(personal_access_token):
+        raise SeamHttpInvalidTokenError(
+            "A JWT cannot be used as a personal_access_token"
+        )
+
+    if is_client_session_token(personal_access_token):
+        raise SeamHttpInvalidTokenError(
+            "A Client Session Token cannot be used as a personal_access_token"
+        )
+
+    if is_publishable_key(personal_access_token):
+        raise SeamHttpInvalidTokenError(
+            "A Publishable Key cannot be used as a personal_access_token"
+        )
+
+    if not is_access_token(personal_access_token):
+        raise SeamHttpInvalidTokenError(
+            f"Unknown or invalid personal_access_token format, expected token to start with {access_token_prefix}"
+        )
+
+    headers = {"authorization": f"Bearer {personal_access_token}"}
+    if workspace_id is not None:
```

Comment:
We should require this when using PAT

Reply:
It's required [here](https://github.com/seamapi/javascript-http/blob/ca078db6f0231bdaef7ac3f92693ad01d776cb60/src/lib/seam/connect/auth.ts#L175) because of multi workspace auth case?

Reply:
b2f6e45c67270ba52405a373ddc94c39e6815843